packages/router/src/router.ts

```diff
@@ -1232,7 +1234,7 @@ export class Router<
       searchStr,
       state: nextState,
       hash,
-      href: this.history.createHref(`${pathname}${searchStr}${hash}`),
+      href: this.history.createHref(`${pathname}${searchStr}${hashStr}`),
```

Comment:
They are necessary, ref. the last paragraph in the PR description.

`this.state.latestLocation.hash` would sometimes already include a `#` in its value, and the old code would then proceed to add another `#` to the value of `hash`. This would cause ramifications down the line, since the `hash` variable is returned from the current function and used later.

NB! With the new code `hashStr` could still possibly start with `##` I guess, but I haven't observed any errors due to this.